src/west/configuration.py

```diff
@@ -282,7 +291,9 @@ def set(self, option: str, value: Any, configfile: ConfigFile = ConfigFile.LOCAL
         :param value: value to set option to
         :param configfile: type of config file to set the value in
         '''
-
+        paths = self.get_paths(configfile)
+        if len(paths) > 1:
+            raise WestNotFound('Cannot write if multiple configs in use.')
```

Comment:
Simple and smart! Nice.

Reply:
I prefer to have the check at the "lowest level", so keep the check in `_write` at least.

This can be used as a fast return path, or we let it fail later on, both are fine IMO.